src/WebAPI/Routes/LeaderboardRoutes.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Application.Contracts;
+using Domain;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace WebAPI.Routes;
+
+public static class LeaderboardRoutes
+{
+    public static void MapLeaderboardRoutes(this WebApplication app)
+    {
+        var routeGroup = app.MapGroup("").WithTags("Leaderboard");
+
+        routeGroup
+            .MapGet(
+                "/getLeaderboardStats",
+                Ok<IEnumerable<LeaderboardUser>> (HttpContext context) =>
+                {
+                    var rulesContext = context.RequestServices.GetRequiredService<IRulesContext>();
+
+                    //Needs to be universal time in order to work the database date values
+                    var oneMonth = DateTime.Now.AddMonths(-1).ToUniversalTime();
+                    var oneYear = DateTime.Now.AddYears(-1).ToUniversalTime();
+
+                    var result = rulesContext.UserStats
+                        .GroupBy(s => new { s.Name, s.Email })
+                        .Select(group => new LeaderboardUser()
+                        {
+                            Name = group.Key.Name,
+                            Email = group.Key.Email,
+                            LastMonth = group.Count(u => u.Date >= oneMonth),
+                            LastYear = group.Count(u => u.Date >= oneYear),
+                            AllTime = group.Count()
+                        })
+                        .AsEnumerable();
+                    
+                    return TypedResults.Ok(result);
+                }
+            )
```

Comment:
Don't do logic in the presentation layer.

This is a slim CA implementation so move this to a LeaderboardService then inject & return the method here.

e.g.

var output = service.something()
return Ok(output);